infoschema/builder.go

```diff
@@ -74,7 +74,7 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 	// We try to reuse the old allocator, so the cached auto ID can be reused.
 	var allocs autoid.Allocators
 	if tableIDIsValid(oldTableID) {
-		if oldTableID == newTableID && diff.Type != model.ActionRenameTable && diff.Type != model.ActionRebaseAutoID && diff.Type != model.ActionModifyTableAutoIdCache {
+		if oldTableID == newTableID && diff.Type != model.ActionRenameTable && diff.Type != model.ActionRebaseAutoID && diff.Type != model.ActionRebaseAutoRandomBase && diff.Type != model.ActionModifyTableAutoIdCache {
```

Comment:
It seems that if we only rebase `auto_random`, but `auto_id` also rebase(we will clean the cache in TiDB).

Reply:
seems right? should keep the special allocator survive?

Reply:
Could we handle it？This feels confusing to users. And please add some tests.

Reply:
yes, Addressed